### ISSUE TITLE
exit_errmsg/assert_errmsg need a terminating newline

### ISSUE THREAD

**Issue by okeuday** [githwxi/ATS-Postiats#272]
To ensure output is obtained from stderr, it is best to have a terminating newline (i.e., "\n") in the fprintf format of these functions.  The problem may only occur due to `libats/DATS/athread.dats` / `libats/DATS/athread_posix.dats` forcing line-buffered output on execution.  The relevant source code is:

https://github.com/githwxi/ATS-Postiats/blob/45a52c42db8400d5a49f97a99f2e433460b4b157/prelude/CATS/CODEGEN/basics.atxt#L138

https://github.com/githwxi/ATS-Postiats/blob/45a52c42db8400d5a49f97a99f2e433460b4b157/prelude/CATS/CODEGEN/basics.atxt#L169

https://github.com/githwxi/ATS-Postiats/blob/45a52c42db8400d5a49f97a99f2e433460b4b157/prelude/CATS/CODEGEN/basics.atxt#L188

**Comment by githwxi** [comment]
I have made the suggested changes. Thanks!

**Comment by okeuday** [comment]
@githwxi Thanks!  I understand you are focused on ATS3 and it is not my intention to distract you, but I thought I should mention that it would be helpful if `atstype_exnconptr` kept the `string` `(char const *)` from `$mylocation` as a separate field in its struct, so it is always present and then accessible from some function in some form as a `string` result.  I found myself making a function like:

```
    char const * const exn_msg_str = (char const *)exn_msg;
    fprintf(stderr, "exception %s(%d%s%s%s)\n%s\n",
            ((atstype_exnconptr)exn)->exnmsg,
            ((atstype_exnconptr)exn)->exntag,
            exn_msg_str[0] ? ", \"" : "",
            exn_msg_str,
            exn_msg_str[0] ? "\"" : "",
            (char const *)msg);
```
I did this because the `atstype_exnconptr` struct field data isn't accessible in ATS2 from the `exn` view type, from what I saw.  The `exntag` may want to not be exposed, I am not sure, however it is very nice to have access to the `exnmsg` value to know exactly what the `exn` is.  If an ATS2 function existed that would provide the `exnmsg` concatenated with the `$mylocation` (at the location of the `$raise`) in some string format, that would be helpful.  I understand a string argument could be added to any `exn` type you create, however, my concern is related to the `exn` types that already exist in ATS2 because they don't tell you where they come from.  More specifically, this problem appears to be why everyone would always favor `assertloc` instead of `assertexn` because `assertexn` is unable to tell you where the assert happened.
